tests/integration/test_config.py

```diff
@@ -43,17 +46,53 @@ def get_field_value(*names: str, configs: Iterable[ObjectConfig | HostGroupConfi
     )
 
 
+async def get_service_with_config(cluster: Cluster) -> Service:
+    return await cluster.services.get(name__eq="complex_config")
+
+
+async def get_object_config(obj: Service | ConfigHostGroup, httpx_client: AsyncClient) -> dict:
+    url = "/".join(map(str, (*obj.get_own_path(), "configs", "")))
+    response = await httpx_client.get(url=url, params={"ordering": "-id", "limit": 2})
+    assert response.status_code == 200
+
+    for result in response.json()["results"]:
+        if result["isCurrent"]:
+            cfg_id = result["id"]
+            break
+    else:
+        raise RuntimeError("Can't find current config")
+
+    response = await httpx_client.get(url=f"{url}/{cfg_id}/")
+    assert response.status_code == 200
+    assert response.json()["isCurrent"]
+
+    return response.json()["config"]
+
+
+async def refresh_and_get_two_configs(
+    obj1: Service | ConfigHostGroup, obj2: Service | ConfigHostGroup
+) -> tuple[ObjectConfig | HostGroupConfig, ObjectConfig | HostGroupConfig]:
+    await obj1.refresh()
+    await obj2.refresh()
+
+    cfg1 = await obj1.config
+    cfg2 = await obj2.config
+
+    if type(obj1) is type(obj2) and obj1.id == obj2.id:
+        assert cfg1.data.id == cfg2.data.id
+        assert cfg1._parent._data["id"] == cfg2._parent._data["id"]  # pyright: ignore[reportAttributeAccessIssue]
+    assert id(cfg1._parent._requester) != id(cfg2._parent._requester)  # pyright: ignore[reportAttributeAccessIssue]
+
+    return cfg1, cfg2
```

Comment:
why so complex logic with checks and ifs?

maybe this is enough:
```python
async def refresh_and_retrieve_config(o: withconfig) -> oc|hgc:
    await o.refresh()
    return await o.config
```

or the same for multiple inputs with yield

Reply:
refactored